persistence/src/main/java/edu/softserve/zoo/persistence/specification/hibernate/impl/house/HouseGetAllBySpeciesIdSpecification.java

```diff
@@ -0,0 +1,35 @@
+package edu.softserve.zoo.persistence.specification.hibernate.impl.house;
+
+import edu.softserve.zoo.exceptions.ApplicationException;
+import edu.softserve.zoo.model.House;
+import edu.softserve.zoo.model.Species;
+import edu.softserve.zoo.persistence.exception.SpecificationException;
+import edu.softserve.zoo.persistence.specification.hibernate.HQLSpecification;
+import edu.softserve.zoo.util.Validator;
+
+/**
+ * House specification for retrieving all houses that applies to a specific species id
+ *
+ * @author Serhii Alekseichenko
+ */
+public class HouseGetAllBySpeciesIdSpecification implements HQLSpecification<House> {
+    private final Long speciesId;
+
+    /**
+     * @param speciesId {@link Species} identifier to filter houses
+     */
+    public HouseGetAllBySpeciesIdSpecification(Long speciesId) {
+        this.speciesId = speciesId;
```

Comment:
validation was moved to the constructor, so exception will be thrown at this stage